python/cudf/cudf/core/dataframe.py

```diff
@@ -340,10 +340,7 @@ def __getitem__(self, arg):
             df = DataFrame()
             if mask.dtype == "bool":
                 # New df-wide index
-                selvals, selinds = column.column_select_by_boolmask(
-                    column.as_column(self.index), Series(mask)
-                )
-                index = self.index.take(selinds.to_gpu_array())
+                index = self.index.take(Series(mask))
```

Comment:
Should we use a column instead of a Series here? We shouldn't need an Index for a gather / bool mask operation.
```suggestion
                index = self.index.take(as_column(mask))
```

Reply:
👍